cli/azd/pkg/auth/manager.go

```diff
@@ -622,8 +643,29 @@ type persistedSecret struct {
 	// base64 string.
 	ClientCertificate *string `json:"clientCertificate,omitempty"`
 
-	// The federated client credential.
-	FederatedToken *string `json:"federatedToken,omitempty"`
+	// The federated auth credential.
+	FederatedAuth *federatedAuth `json:"federatedAuth,omitempty"`
+}
+
+// federated auth token providers
+var (
+	gitHubFederatedAuth federatedTokenProvider = "github"
+)
+
+// token provider for federated auth
+type federatedTokenProvider string
+
+// federatedAuth stores federated authentication information.
+//
+// federatedAuth operates under two distinct modes:
+//   - When Token is set, the static token (which may have limited lifetime) is used for authentication.
+//   - When Provider is set, a token is obtained by calling the provider as needed.
+type federatedAuth struct {
```

Comment:
I think this could be a breaking change in the case where some existing user has logged in using `--federated-token`, since the shape of the stored data has changed in config, previously it was a string and now it is an object.  I think in practice this is unlikely to hit anyone, because the configuration data was likely stored on an ephemeral machine (like a CI runner) and so there's no concern, but I wanted to call it out.

Part of me wonders if we should just remove the support for using a fixed federated token and just say that only `--federated-token-provider` is supported.  I think in practice these federated tokens will be short lived (since the whole point of using them is to get rid of long-lived secrets), so we'd still run into this root issue for anyone who logs in using a fixed token.

Interested in your thoughts on this, @weikanglim.

Reply:
@ellismg You're correct that this is indeed breaking, and the user would have to run `azd auth login` again if they were impacted by it. However, as the token validity is not more than 5 minutes, and as you mentioned, the configuration is thrown away at the end of a true CI execution, I don't think we'll run afoul of any users.

I've been wondering about the efficacy of the token feature support as well. I agree with removing the fixed token support.

Reply:
> I don't think we'll run afoul of any users.

Yup.

> I've been wondering about the efficacy of the token feature support as well. I agree with removing the fixed token support.

SGTM

Reply:
> Are we okay with keeping the schema for persistedSecret as the following?

Yes - this seems good.  I agree that the ability to provide additional options will be useful long term, so it does seem worthwhile to make this thing an object.